src/Symfony/Component/Filesystem/Path.php

```diff
@@ -0,0 +1,990 @@
+     *
+     * ```php
+     * echo Path::makeRelative("/webmozart/style.css", "/webmozart/puli");
+     * // => ../style.css
+     * ```
+     *
+     * If a relative path is passed and the base path is absolute, the relative
+     * path is returned unchanged:
+     *
+     * ```php
+     * Path::makeRelative("style.css", "/webmozart/puli/css");
+     * // => style.css
+     * ```
+     *
+     * If both paths are relative, the relative path is created with the
+     * assumption that both paths are relative to the same directory:
+     *
+     * ```php
+     * Path::makeRelative("style.css", "webmozart/puli/css");
+     * // => ../../../style.css
+     * ```
+     *
+     * If both paths are absolute, their root directory must be the same,
+     * otherwise an exception is thrown:
+     *
+     * ```php
+     * Path::makeRelative("C:/webmozart/style.css", "/webmozart/puli");
+     * // InvalidArgumentException
+     * ```
+     *
+     * If the passed path is absolute, but the base path is not, an exception
+     * is thrown as well:
+     *
+     * ```php
+     * Path::makeRelative("/webmozart/style.css", "webmozart/puli");
+     * // InvalidArgumentException
+     * ```
+     *
+     * If the base path is not an absolute path, an exception is thrown.
+     *
+     * The result is a canonical path.
+     *
+     * @param string $path     A path to make relative.
+     * @param string $basePath A base path.
+     *
+     * @return string A relative path in canonical form.
+     *
+     * @throws InvalidArgumentException If the base path is not absolute or if
+     *                                  the given path has a different root
+     *                                  than the base path.
+     *
+     * @since 1.0 Added method.
+     * @since 2.0 Method now fails if $path or $basePath is not a string.
+     */
+    public static function makeRelative(string $path, string $basePath): string
+    {
+        $path = static::canonicalize($path);
+        $basePath = static::canonicalize($basePath);
+
+        [$root, $relativePath] = self::split($path);
+        [$baseRoot, $relativeBasePath] = self::split($basePath);
+
+        // If the base path is given as absolute path and the path is already
+        // relative, consider it to be relative to the given absolute path
+        // already
+        if ('' === $root && '' !== $baseRoot) {
+            // If base path is already in its root
+            if ('' === $relativeBasePath) {
+                $relativePath = ltrim($relativePath, './\\');
+            }
+
+            return $relativePath;
+        }
+
+        // If the passed path is absolute, but the base path is not, we
+        // cannot generate a relative path
+        if ('' !== $root && '' === $baseRoot) {
+            throw new InvalidArgumentException(sprintf(
+                'The absolute path "%s" cannot be made relative to the '.
+                'relative path "%s". You should provide an absolute base '.
```

Comment:
strings for exception messages should stay in a single string rather than being split by concatenation, to follow our coding standards (it makes it much easier to search for an error message in the codebase).

Reply:
Done, checked for the other occurrences as well